Mathlib/CategoryTheory/DifferentialObject.lean

```diff
@@ -33,75 +33,96 @@ namespace CategoryTheory
 
 variable (C : Type u) [Category.{v} C]
 
--- TODO: generalize to `has_shift C A` for an arbitrary `[add_monoid A]` `[has_one A]`.
+-- TODO: generalize to `HasShift C A` for an arbitrary `[AddMonoid A]` `[One A]`.
 variable [HasZeroMorphisms C] [HasShift C ℤ]
 
 /-- A differential object in a category with zero morphisms and a shift is
-an object `X` equipped with
-a morphism `d : X ⟶ X⟦1⟧`, such that `d^2 = 0`.
--/
-@[nolint has_nonempty_instance]
+an object `pt` equipped with
+a morphism `d : X ⟶ X⟦1⟧`, such that `d^2 = 0`. -/
+-- Porting note: Removed `@[nolint has_nonempty_instance]`
 structure DifferentialObject where
+  /-- The underlying object of a differential object. -/
   pt : C
-  d : X ⟶ X⟦1⟧
-  d_squared' : d ≫ d⟦(1 : ℤ)⟧' = 0 := by obviously
+  /-- The differential of a differential object. -/
+  d : pt ⟶ pt⟦(1 : ℤ)⟧
+  /-- The differential `d` satisfies that `d² = 0`. -/
+  d_squared' : d ≫ d⟦(1 : ℤ)⟧' = 0 := by aesop_cat -- Porting note: Originally `by obviously`
 #align category_theory.differential_object CategoryTheory.DifferentialObject
 
-restate_axiom differential_object.d_squared'
+-- Porting note : this resulted in a `INCORRECT DEF/LEMMA` liner error, so `Hom.comm` is
+-- made manually
+-- restate_axiom DifferentialObject.d_squared'
 
-attribute [simp] differential_object.d_squared
+-- attribute [simp] DifferentialObject.d_squared
 
 variable {C}
 
 namespace DifferentialObject
 
-/-- A morphism of differential objects is a morphism commuting with the differentials.
--/
-@[ext, nolint has_nonempty_instance]
+@[simp]
+lemma d_squared (X : DifferentialObject C) : X.d ≫ X.d⟦1⟧' = 0 := X.d_squared'
+
+/-- A morphism of differential objects is a morphism commuting with the differentials. -/
+@[ext] -- Porting note: Removed `nolint has_nonempty_instance`
 structure Hom (X Y : DifferentialObject C) where
+  /-- the morphism between underlying objects of the two differentiable object -/
```

Comment:
I'm not averse to adding documentation, if it's done well. (Realistically we're comparing adding it now or not at all, rather than nor or later.)

However doc-strings should be complete senetences, so:
```suggestion
  /-- The morphism between underlying objects of the two differentiable objects. -/
```